perun-core/src/main/java/cz/metacentrum/perun/core/blImpl/AttributesManagerBlImpl.java

```diff
@@ -1442,6 +1593,44 @@ public boolean setAttributeWithoutCheck(PerunSession sess, Resource resource, Me
 		return changed;
 	}
 
+	@Override
+	public boolean setAttributeWithoutCheck(PerunSession sess, Member member, Group group, Attribute attribute, boolean workWithUserAttributes) throws InternalErrorException, WrongAttributeAssignmentException, WrongAttributeValueException, WrongReferenceAttributeValueException {
+		if(getAttributesManagerImpl().isCoreAttribute(sess, attribute)) throw new WrongAttributeAssignmentException(attribute);
+
+		boolean changed = true;
+		if(getAttributesManagerImpl().isFromNamespace(sess, attribute, AttributesManager.NS_MEMBER_GROUP_ATTR)) {
+			if(getAttributesManagerImpl().isVirtAttribute(sess, attribute)) {
+				throw new InternalErrorException("Virtual attribute can't be set this way yet. Please set physical attribute instead.");
```

Comment:
I prefer to create method which be able to do it like getAttributesManagerImpl().setVirtualAttribute(sess, member, group, attribute);